formula/src/main/java/com/instacart/formula/FormulaRuntime.kt

```diff
@@ -190,61 +187,62 @@ class FormulaRuntime<Input : Any, Output : Any>(
                  */
                 isRunEnabled = true
 
-                if (globalEffectQueue.isNotEmpty() || pendingEvaluation) {
-                    val evaluate = pendingEvaluation
-                    pendingEvaluation = false
-                    run(evaluate = evaluate)
-                }
+                runIfNeeded()
             }
         }
     }
 
+    private fun runIfNeeded() {
+        if (globalEffectQueue.isNotEmpty() || pendingEvaluation) {
+            val evaluate = pendingEvaluation
+            pendingEvaluation = false
+            run(evaluate = evaluate)
+        }
+    }
+
     /**
      * Performs the evaluation and execution phases.
      */
     private fun run(evaluate: Boolean = true) {
         if (isRunning) return
 
         try {
-            val manager = checkNotNull(manager)
+            val manager = requireManager()
 
             if (evaluate) {
                 var shouldRun = true
                 while (shouldRun) {
                     val localInputId = inputId
-                    if (!manager.isTerminated()) {
```

Comment:
It's not possible to hit this case because `evaluate` will be false if the manager is terminated